### ISSUE TITLE
Migrate umbraco.library from wiki to docs

### ISSUE THREAD

**Issue by hemraker** [umbraco/UmbracoDocs#280]
Wiki page can be found here https://our.umbraco.org/wiki/reference/umbracolibrary/

New umbraco.library docs should be available here https://our.umbraco.org/documentation/Reference/Templating/Macros/Xslt/UmbracoLibrary/
and redirect from documentation/reference/api/umbracolibrary should be updated to reflect this

Work will be done in this branch https://github.com/umbraco/UmbracoDocs/tree/UmbracoLibrary

**Comment by hemraker** [comment]
@greystate Initial commit 1d38e880f69932c3f6108c84199f98fe71a644ff

**Comment by greystate** [comment]
Cüül! :metal:

**Comment by greystate** [comment]
I think it should be enough with a single Index.md file in that folder, documenting the various extensions. It should only serve as a reference for those who've got their hands on an older solution built with XSLT, that they need to do some reference lookups for.

So if it's modeled like the UmbracoHelper page it should be easy to follow, I guess?

**Comment by hemraker** [comment]
Could do that, although some of the examples are quite extensive and it would require a lot of scrolling. Also it shouldn't be for reference for  old versions (that's what the wiki is for) instead correct docs on using Umbraco.Library today.

**Comment by greystate** [comment]
Sounds great - I'll proceed as planned :-)

**Comment by greystate** [comment]
This just to let @hemraker know that there's progress here :)

[Here's a comparison][COMPARE] - and you can [browse them here][BROWSE] and see if this is the way to proceed...

[COMPARE]: https://github.com/umbraco/UmbracoDocs/compare/UmbracoLibrary...greystate:UmbracoLibrary

[BROWSE]: https://github.com/greystate/Umbraco4Docs/blob/UmbracoLibrary/Reference/Templating/Macros/Xslt/UmbracoLibrary/index.md

**Comment by sofietoft** [comment]
@hemraker,
I'm having a bit of a hard time figuring this one out.
The branch is still there: https://github.com/umbraco/UmbracoDocs/tree/UmbracoLibrary

But this documentation/reference/api/umbracolibrary , redirects to https://our.umbraco.org/documentation/Reference/Routing/WebApi/
And this https://our.umbraco.org/documentation/Reference/Templating/Macros/Xslt/UmbracoLibrary/ doesn't exist.

Help me out here? 😁 

-Sofie

**Comment by dampee** [comment]
Hi,
The Library is now the [UmbracoHelper](https://our.umbraco.org/Documentation/Reference/Querying/UmbracoHelper/). I guess we can just close this issue.

Or maybe ask @nul800sebastiaan  why documentation/reference/api/umbracolibrary , redirects to https://our.umbraco.org/documentation/Reference/Routing/WebApi/... but other than that: close it!

**Comment by marcemarc** [comment]
I've added this page: https://our.umbraco.org/documentation/Reference/Templating/Macros/Xslt/Umbraco.Library/
(for a different issue - to get rid of all pages that say 'documentation is waiting')

It just lists the methods of Umbraco.Library but without any detail...

Is this enough to draw a close to this issue too?

Is there anything else valuable in the wiki: https://en.wikibooks.org/wiki/Umbraco/Reference/umbraco.library that it would be beneficial to migrate here...

and should @hemraker's Nice Url from here:

https://github.com/umbraco/UmbracoDocs/blob/UmbracoLibrary/Reference/Templating/Macros/Xslt/UmbracoLibrary/NiceUrl/index.md

be merged in and linked from the page I've added?

or are we feeling we've missed the opportunity to document these methods fully? but this minimum announcing of their presence is enough?
